movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx

```diff
@@ -0,0 +1,79 @@
+import {Button} from "../../components/Button/Button";
+import {useTranslation} from "react-i18next";
+
+type MovieProps = {
+    title: string | '';
+    description: string | '';
+    key: string | '';
+}
+
+type MovieFormProps = {
+    onCloseModal: Function;
+    onAddMovie: Function;
+    currentValue:MovieProps;
+}
+
+export function AddEditMovieForm(props:MovieFormProps): JSX.Element {
+    const {t} = useTranslation();
+    const currentValueToEdit = props.currentValue;
+
+    const submitForm = (e:any) => {
+        const title = e.target.elements.movieTitle.value;
+        const description = e.target.elements.movieDescription.value;
+        let key = e.target.elements.movieKey.value;
+
+        if (!key) {
+            key = `${title}_${description}_${Math.random().toString()}`;
+            props.onAddMovie((prev: any) => {
+                return [...prev, {key, title, description}];
+            });
+        } else {
+            props.onAddMovie((prev: any) => {
+                return prev.map((el:any) => el.key === key ? {key, title, description} : el);
+            });
+        }
+        props.onCloseModal();
+    }
+
+    const closeModal = () => {
+        props.onCloseModal();
+    }
+
+    return (
+        <>
+            <h2>{t('add_form_title')}</h2>
+            <Button
+                type={'button'}
+                onclickAction={closeModal}
+                title={t('label_close')}
```

Comment:
about naming convention - snake case for variables is usually used at Python or backend applications in averall, but it's up to u) just small remark - u can leave as u want